src/formatDate.js

```diff
@@ -50,7 +50,66 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = {
+    separator: fromFormat[fromFormat.length - 1],
+  };
+
+  const separateDate = date.split(dateParts.separator);
+
+  for (const partFrom of fromFormat) {
+    switch (partFrom) {
+      case 'YY':
+        dateParts.year = separateDate[fromFormat.indexOf(partFrom)];
```

Comment:
Since you need indexes and indexOf(partFrom) is repeating , using for loop with  indexes could be a better idea.